packages/cli/src/UserManagement/decorators.ts

```diff
@@ -0,0 +1,89 @@
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+/* eslint-disable @typescript-eslint/naming-convention */
+/* eslint-disable import/no-cycle */
+import { Application, Request, RequestHandler as Middleware, Response, Router } from 'express';
+import { posix } from 'path';
+import { ResponseHelper } from '..';
+import config from '../../config';
+
+interface RequestHandler<ReqBody, ResBody> extends Function {
+	async(req: Request<unknown, ResBody, ReqBody>, res: Response<ResBody>): Promise<ReqBody>;
+}
+
+export interface IRouter {
+	method: Method;
+	path: string;
+	handlerName: string | symbol;
+}
+
+const Keys = {
+	ROUTES: Symbol('routes'),
+	BASE_PATH: Symbol('base_path'),
+} as const;
+
+export enum Method {
+	GET = 'get',
+	POST = 'post',
+	PATCH = 'patch',
+	DELETE = 'delete',
+}
+
+export const RestController =
+	(basePath = '/'): ClassDecorator =>
```

Comment:
I'm using `require('path').posix.join` to form the final urls, so the leading `/` isn't actually mandatory.
but if you still think it'd be good to enforce the leading slash, then let's do it.

Reply:
`posix.join` optimizes for the writer instead of for the readers, who can be confused by all the inconsistencies we're allowing and silently autofixing in routes. This also leads to lower searchability. Probably not too much of a burden for the writer to specify the correct endpoint without implicit aid.

Reply:
okay, then I'll get rid of `posix.join` and switch to enforcing a leading slash, as you suggested.